src/libraries/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -971,6 +803,12 @@ private void MergeUpdateCollections(ref CookieCollection? destination, CookieCol
                             to_add = false;
                         }
 
+                        // For implicit domains exact match is needed
+                        if (cookie.DomainImplicit && !string.Equals(host, cookie.Domain, StringComparison.OrdinalIgnoreCase))
```

Comment:
This is the thing that makes [GetCookies_AddCookiesWithImplicitDomain_CookiesReturnedOnlyForExactDomainMatch](https://github.com/dotnet/runtime/blob/58b068bea2c150d19ff642d751e6111a50f05c33/src/libraries/System.Net.Primitives/tests/FunctionalTests/CookieTest/CookieContainerTest.cs#L30-L62) pass, currently it is just an accident (the domains passed to the method `BuildCookieCollectionFromDomainMatches` mismatch the host because of a leading dot difference.)